### PR TITLE
Improve error and warning messages

### PR DESCRIPTION
**Pull request by ybiquitous**
> Explain a summary, purpose, or background for this change.

- Simplify the messages.
- Fix incorrect document URLs.
- Make the format consistent.

> Link related issues, e.g. "Fix #<ISSUE_ID>", "Related to #<ISSUE_ID>", or "None."

None.

> Check the following items.

- [x] Add a new [changelog](https://github.com/sider/runners/blob/master/CHANGELOG.md) entry if this change is notable.

### CODE REVIEW

File: test/config_test.rb [hunk 1]
```diff
@@ -110,7 +110,7 @@ def test_content_with_unknown_linter
       exn = assert_raises Runners::Config::InvalidConfiguration do
         Runners::Config.new(path)
       end
-      assert_equal "The attribute `$.linter.unknown_linter` in your `sider.yml` is unsupported. Please fix and retry.", exn.message
+      assert_equal "The attribute `linter.unknown_linter` in your `sider.yml` is unsupported. Please fix and retry.", exn.message
```
Comment (ybiquitous) [inline test/config_test.rb:113]: [note] Make consistent with the document: `$.linter` -> `linter`
https://help.sider.review/getting-started/custom-configuration

**Comment by ybiquitous** [comment]
@yykamei What do you think about this change? Any suggestions?

File: lib/runners/processor.rb [hunk 2]
```diff
@@ -190,8 +191,8 @@ def add_warning_for_deprecated_linter(alternative:, ref:, deadline: nil)
       deadline_str = deadline ? deadline.strftime("on %B %-d, %Y") : "in the near future"
       add_warning <<~MSG, file: config.path_name
         DEPRECATION WARNING!!!
-        The support for #{analyzer_name} is deprecated. Sider will drop these versions #{deadline_str}.
-        Please consider using an alternative tool #{alternative}. See #{ref}
+        The support for #{analyzer_name} is deprecated and will be removed #{deadline_str}.
+        Please migrate to #{alternative} which is an alternative. See #{ref}
```
[the remark this answers is not in the document]
Comment (ybiquitous) [reply lib/runners/processor.rb:195]: Um..., both look the same meanings. 🤔

File: lib/runners/processor.rb [hunk 3]
```diff
@@ -167,21 +167,22 @@ def add_warning_if_deprecated_version(minimum:, file: nil, deadline: nil)
         deadline_str = deadline ? deadline.strftime('on %B %-d, %Y') : 'in the near future'
         add_warning <<~MSG, file: file
           DEPRECATION WARNING!!!
-          The #{analyzer_version} and older versions are deprecated. Sider will drop these versions #{deadline_str}.
+          The #{analyzer_version} and older versions are deprecated and these versions will be dropped #{deadline_str}.
```
[the remark this answers is not in the document]
Comment (ybiquitous) [reply lib/runners/processor.rb:170]: Good catch!

File: lib/runners/processor.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (ybiquitous) [reply lib/runners/processor.rb:195]:
```suggestion
        Please migrate to #{alternative} as an alternative. See #{ref}
```

Maybe the above is more clear?

[the remark this answers is not in the document]
Comment (ybiquitous) [reply lib/runners/processor.rb]: Fixed.

**Comment by ybiquitous** [comment]
Thank you so much for the review! I've addressed your comments.
When the CI passes, I'll merge this.

File: lib/runners/processor.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (yykamei) [reply lib/runners/processor.rb]: From "ESLint which is an alternative," I feel it implies there are other ESLint that are not alternatives.
From "ESLint, which is an alternative," I think that the clause just explains ESLint.
Though I like ", which", using "as" might be clear too.

**Pull request merged**